boot/zephyr/boot_serial_extension_zephyr_basic.c

```diff
@@ -0,0 +1,166 @@
+    } else {
+        rc = flash_area_erase(fa, 0, flash_area_get_size(fa));
+        if (rc < 0) {
+            BOOT_LOG_ERR("failed to erase flash area");
+        }
+        flash_area_close(fa);
+    }
+    if (rc == 0) {
+        rc = MGMT_ERR_OK;
+    } else {
+        rc = MGMT_ERR_EUNKNOWN;
+    }
+
+    zcbor_map_start_encode(cs, 10);
+    zcbor_tstr_put_lit(cs, "rc");
+    zcbor_uint32_put(cs, rc);
+    zcbor_map_end_encode(cs, 10);
+
+    return rc;
+}
+
+MCUMGR_HANDLER_DEFINE(storage_erase, bs_custom_storage_erase);
+#endif
+
+#ifdef MCUBOOT_MGMT_CUSTOM_IMG_LIST
+static int custom_img_status(int image_index, uint32_t slot, char *buffer,
+                             ssize_t len)
+{
+    uint32_t area_id;
+    struct flash_area const *fap;
+    struct image_header hdr;
+    int rc;
+    int img_install_stat;
+
+    rc = BOOT_HOOK_CALL(boot_img_install_stat_hook, BOOT_HOOK_REGULAR,
+                        image_index, slot, &img_install_stat);
+    if (rc == BOOT_HOOK_REGULAR)
+    {
+        img_install_stat = 0;
+    }
+
+    rc = BOOT_HOOK_CALL(boot_read_image_header_hook, BOOT_HOOK_REGULAR,
+                        image_index, slot, &hdr);
+    if (rc == BOOT_HOOK_REGULAR)
+    {
+        area_id = flash_area_id_from_multi_image_slot(image_index, slot);
+
+        rc = flash_area_open(area_id, &fap);
+        if (rc) {
+            return rc;
+        }
+
+        rc = flash_area_read(fap, 0, &hdr, sizeof(hdr));
+
+        flash_area_close(fap);
+    }
+
+    if (rc == 0) {
+        if (hdr.ih_magic == IMAGE_MAGIC) {
+            snprintf(buffer, len, "ver=%d.%d.%d.%d,install_stat=%d",
+                    hdr.ih_ver.iv_major,
+                    hdr.ih_ver.iv_minor,
+                    hdr.ih_ver.iv_revision,
+                    hdr.ih_ver.iv_build_num,
+                    img_install_stat);
+        } else {
+            rc = 1;
+        }
+    }
+
+    return rc;
+}
+
+static int bs_custom_img_list(const struct nmgr_hdr *hdr, const char *buffer,
+                              int len, zcbor_state_t *cs)
+{
+    int rc = 0;
+    char tmpbuf[64];	/* Buffer should fit version and flags */
+
+    (void)buffer;
+    (void)len;
+
+    if (hdr->nh_group != ZEPHYR_MGMT_GRP_BASIC || hdr->nh_op != NMGR_OP_READ ||
+        hdr->nh_id != ZEPHYR_MGMT_GRP_BASIC_CMD_IMAGE_LIST) {
+        return MGMT_ERR_ENOTSUP;
+    }
+
+    zcbor_map_start_encode(cs, 10);
+
+    for (int img = 0; img < MCUBOOT_IMAGE_NUMBER; img++) {
+        for (int slot = 0; slot < 2; slot++) {
+            rc = custom_img_status(img, slot, tmpbuf, sizeof(tmpbuf));
+
+            zcbor_int32_put(cs, img * 2 + slot + 1);
+            if (rc == 0) {
+                zcbor_tstr_put_term(cs, tmpbuf);
+            } else {
+                zcbor_tstr_put_lit(cs, "");
+            }
+        }
+    }
+
+    zcbor_tstr_put_lit(cs, "rc");
+    zcbor_uint32_put(cs, MGMT_ERR_OK);
+    zcbor_map_end_encode(cs, 10);
+
+    return rc;
```

Comment:
I am not sure whether `rc` here is stable: in case when it fails on the first slot in loop, then we have rc == 0 here finally and it seems that we return success even though we have failed in the first iteration, but if we fail in the last iteration then we return error here.
Does it make sense to return here something else than 0 if we assume that in case of error we just return empty string? Or should this be reflected by the "rc" returned by line 152?

Reply:
You know you are right in that it makes no sense, though it does appear that this code is actually what it does now: https://github.com/mcu-tools/mcuboot/blob/main/boot/zephyr/boot_serial_extensions.c

Not really sure what this should return, I don't see this command documented in the zephyr docs?

Reply:
In fact I would consider just removing this command entirely, what is the point in it? This is the hook it calls:
```
int boot_img_install_stat_hook(int image_index, int slot, int *img_install_stat)
{
    return BOOT_HOOK_REGULAR;
}
```
So why return this at all then img mgmt list can return actual information including status about the image

Reply:
@nvlsianpu since you added this, what is the point in this command? Does anything use it? Can we just get rid of it?

Reply:
This command was added in order to make possible (among other) fetch NET core image installation status on nRF53, which was never used anywhere.

Reply:
Regard the return value - I would make it more consistent with failures this code could encounter.

Reply:
I'm of the mind it should just be deleted, if something is needed downstream for this it can be added downstream, and this is really not at all in the style of MCUmgr commands by generating a string with multiple pieces of information - in CSV format, and just appending it. I will update the PR